pulsar-client-admin/src/main/java/com/yahoo/pulsar/client/admin/internal/PersistentTopicsImpl.java

```diff
@@ -297,6 +299,38 @@ public void failed(Throwable throwable) {
     }
 
     @Override
+    public JsonObject getInternalInfo(String destination) throws PulsarAdminException {
+        try {
+            return getInternalInfoAsync(destination).get();
+        } catch (ExecutionException e) {
+            throw (PulsarAdminException) e.getCause();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new PulsarAdminException(e.getCause());
+        }
+    }
+
+    @Override
+    public CompletableFuture<JsonObject> getInternalInfoAsync(String destination) {
+        DestinationName ds = validateTopic(destination);
+        final CompletableFuture<JsonObject> future = new CompletableFuture<>();
+        asyncGetRequest(persistentTopics.path(ds.getNamespace()).path(ds.getEncodedLocalName()).path("internal-info"),
+                new InvocationCallback<String>() {
```

Comment:
The reason for passing a JSONObject rather than a POJO is to avoid having the client depending on `managed-ledger` which is where the POJO is defined.